src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,972 @@
+                var currentInvocationArguments = syntaxFacts.GetArgumentsOfArgumentList(currentArgumentListSyntax);
+                var requiredArguments = new List<SyntaxNode>();
+
+                foreach (var parameterSymbol in validParameters)
+                {
+                    if (parameterToArgumentMap.TryGetValue(parameterSymbol, out var index))
+                    {
+                        requiredArguments.Add(currentInvocationArguments[index]);
+                    }
+                }
+
+                var conditionalRoot = syntaxFacts.GetRootConditionalAccessExpression(invocation);
+                var named = ShouldArgumentBeNamed(compilation, semanticFacts, invocationSemanticModel, invocationArguments, insertionIndex, cancellationToken);
+                var newMethodInvocation = GenerateNewMethodInvocation(syntaxFacts, generator, invocation, requiredArguments, newMethodIdentifier);
+
+                SeparatedSyntaxList<SyntaxNode> allArguments;
+                if (conditionalRoot is null)
+                {
+                    allArguments = AddArgumentToArgumentList(currentInvocationArguments, generator, newMethodInvocation, parameterName, insertionIndex, named);
+                }
+                else
+                {
+                    var expressionsWithConditionalAccessors = conditionalRoot.ReplaceNode(invocation, newMethodInvocation);
+                    allArguments = AddArgumentToArgumentList(currentInvocationArguments, generator, expressionsWithConditionalAccessors, parameterName, insertionIndex, named);
+                }
+
+                return UpdateArgumentListSyntax(currentArgumentListSyntax, allArguments);
+            }
+        }
+
+        private static SyntaxNode GenerateNewMethodInvocation(ISyntaxFactsService syntaxFacts, SyntaxGenerator generator, SyntaxNode invocation, List<SyntaxNode> arguments, string newMethodIdentifier)
+        {
+            var methodName = generator.IdentifierName(newMethodIdentifier);
+            var fullExpression = syntaxFacts.GetExpressionOfInvocationExpression(invocation);
+            if (syntaxFacts.IsAnyMemberAccessExpression(fullExpression))
+            {
+                var identifierExpression = syntaxFacts.GetExpressionOfMemberAccessExpression(fullExpression);
+                if (identifierExpression is not null)
+                {
+                    methodName = generator.MemberAccessExpression(identifierExpression, newMethodIdentifier);
+                }
+                else
+                {
+                    // In the VB case sometimes the the member access expression's expression is nothing
+                    // so I generate a member binding expression instead.
+                    methodName = generator.MemberBindingExpression(generator.IdentifierName(newMethodIdentifier));
```

Comment:
this happens in a VB with block.  i.e.

```
With x
    .Foo(1, 2, 3)
End With
```

Do you have tests for this?

Reply:
i think it should still be a MemberAccessExpression, just with a null-expression

Reply:
the generator just creates an member access expression if the generation is being invoked in VB. It just calls the SyntaxFactory, but since that is language-specific, I was trying to avoid doing that myself. Passing in null to the generators MemberAccessExpression throws an error

Reply:
that's a bug with teh vb generator.  it should allow for that :) i'm ok with you fixing that as part of this pr.